### ISSUE TITLE
[bug] Not working for nested component inside component

### ISSUE THREAD

**Issue by GorvGoyl** [joao-mbn/react-extract#12]
The "React Extract" action is not available when I select JSX from a nested component. I have many nested components that I would like to extract, but this extension is not working in those cases.

Example:

```js
export function VerifyEmail() {
    const [name, setName] = useState('');

    function Welcome() {
        return (
            <div className="bg-neutral-100">
                <div className="rounded border">Hello, {name}</div>
            </div>
        );
    }

    return (
        <div className="text-lg text-black">
            <Welcome />
            <div className="text-neutral-500">Please verify your email to continue</div>
        </div>
    );
}
```

![screenshot 2024-07-08 at 13 28 59@2x](https://github.com/joao-mbn/react-extract/assets/7106086/853e69fb-d94e-46bc-9c42-0e60532d5f06)

**Comment by joao-mbn** [comment]
Hi @GorvGoyl, thanks for reporting this bug. 

I believe I have overlooked this scenario. I'll keep this thread updated according to my findings and fixes.

**Comment by joao-mbn** [comment]
Fixed in 757002a and made available in `v0.10.0`.

In fact, this was not a bug, but a bad feature: early in this project I decided to use the vscode diagnostics as a way to filter out unintended selections. But it might well be the case that someone wants to extract a code with warnings or errors and that should ultimately be the user's the decision. In `v0.10.0` that is the case.
